qiling/core_utils.py

```diff
@@ -46,13 +44,14 @@ def nprint(self, *args, **kw):
 
             # setup filter for logger
             # FIXME: only works for logging due to we might need runtime disable nprint, it should be a global filter not only syscall
-            # if self.strace != None and self.output == QL_OUTPUT.DEFAULT:
-            #     from .utils import Ql_filter
-            #     self.log_file_fd.addFilter(ql_filter(self.filter))
+            if self.filter != None and self.output == QL_OUTPUT.DEFAULT:
+                from .utils import ql_setup_filter
+                self.log_file_fd.addFilter(ql_setup_filter(self.filter))
 
             msg = args[0]
             msg += kw["end"] if kw.get("end", None) != None else os.linesep
-            fd.info(msg)
+            if self.console == True:
+                fd.info(msg)
```

Comment:
One more question. 

if ql.console == False will it still write to the log ?

Reply:
I guess this is the last reqeust

Once we confirm this we are good to go.

Reply:
done it by add a FalseFilter for all stream handler when console=False, at the same time still write to disk logging file

Reply:
testing code:
```
#!/usr/bin/env python3
# 
# Cross Platform and Multi Architecture Advanced Binary Emulation Framework
# Built on top of Unicorn emulator (www.unicorn-engine.org) 
import sys
sys.path.append("..")
from qiling.os.posix.syscall.fcntl import ql_syscall_openat
from qiling import *

def my_openat(ql, *args):
    ql.console = True
    ql.nprint("Shows something")
    ql.console = False
    ql.nprint("Shows nothing") 
    ql_syscall_openat(ql, *args)

if __name__ == "__main__":
    ql = Qiling(["examples/rootfs/x8664_linux/bin/x8664_hello"], "examples/rootfs/x8664_linux", console=False, log_dir='qlog')
    ql.set_syscall("openat", my_openat)
    ql.run()
```

Reply:
result from console:
![image](https://user-images.githubusercontent.com/10076774/82137606-44f40280-984c-11ea-9fcc-a68c4f1d29e9.png)

Reply:
result from log:
![image](https://user-images.githubusercontent.com/10076774/82137619-5b01c300-984c-11ea-9869-459f14fa5225.png)